File: 0163-missing-ranges/0163-missing-ranges.java
class Solution {
    public List<List<Integer>> findMissingRanges(int[] nums, int lower, int upper) {
        List<List<Integer>> answer = new ArrayList<>();
        if(nums.length == 0){
            List<Integer> list = new ArrayList<>();
            list.add(lower);
            list.add(upper);
            answer.add(list);
            return answer;
        }
        if(nums[0]!=lower){
            List<Integer> list = new ArrayList<>();
            list.add(lower);
            list.add(nums[0]-1);
            answer.add(list);
        }

        for(int i=0;i<nums.length - 1; i++){
            if(Math.abs(nums[i]-nums[i+1]) > 1){
                List<Integer> list = new ArrayList<>();
                list.add(nums[i]+1);
                list.add(nums[i+1]-1);
                answer.add(list);
            }
        }
        
        if(nums[nums.length-1]!=upper){
            List<Integer> list = new ArrayList<>();
            list.add(nums[nums.length-1] + 1);
            list.add(upper);
            answer.add(list);
        }

        return answer;
    }
}

File: 0163-missing-ranges/0163-missing-ranges.cs
public class Solution {
    public IList<IList<int>> FindMissingRanges(int[] nums, int lower, int upper) {
        var n = nums.Length;
        var ranges = new List<IList<int>>();
        if(n == 0){
            ranges.Add(new List<int>(){lower, upper});
            return ranges;
        }
        var left = nums[0];
        var right = nums[n - 1];
        

        if(left != lower){
            ranges.Add(new List<int>(){lower, left-1});
        }
        
        for(int i = 0; i < n - 1; i++){
            if(nums[i]+1 == nums[i+1]) continue;
            //found range element
            var newRange = new List<int>(){nums[i]+1, nums[i+1]-1};
            ranges.Add(newRange);
        }
        if(right != upper){
            ranges.Add(new List<int>(){right+1, upper});
        }
        return ranges;
    }
}
/*
[0,1,3,50,75,99]

*/